### PR TITLE
chore(wdio): convert event-custom-type to wdio

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

port the `test/karma/test-app/event-custom-type` test suite to webdriverio.

git history has been preserved as much as possible here, using `git mv` to move files from the `karma` dir to `wdio/` in all possible cases. however, git may fail to mark some files as 'moved' if we change a file enough that it exceeds git's "sameness threshold".

## Documentation

N/A
## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing
To verify this worked, I made the tests "red" (fail) for every assertion, then one by one made them pass/green

## Other information

STENCIL-1191

### CODE REVIEW

File: test/wdio/event-custom-type/cmp.test.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    });
+  });
+
+  it('should dispatch an event on load', async () => {
+    await expect($('#counter')).toHaveText('1');
+  });
+
+  it('should emit a complex type', async () => {
+    // wait to ensure the event it emitted in the component
+    await browser.pause(100);
```
Comment (christian-bromann) [inline test/wdio/event-custom-type/cmp.test.tsx:17]: This should not be needed as WebdriverIO will retry the assertion in case it fails
```suggestion
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]

File: test/wdio/event-custom-type/cmp.test.tsx (hunk 1 above)
Comment (rwaskiewicz) [reply test/wdio/event-custom-type/cmp.test.tsx]: makes sense, thanks!

**Review by christian-bromann — APPROVED** [review]
👍

File: test/wdio/event-custom-type/index.html
Comment (alicewriteswrongs) [inline test/wdio/event-custom-type/index.html:1]: I think maybe you missed deleting the html file here

**Review by alicewriteswrongs — CHANGES_REQUESTED** [review]
oops, meant for that comment to be a review!

**Review by alicewriteswrongs — APPROVED** [review]

**Pull request merged**